resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/Workspace.java

```diff
@@ -1135,7 +1135,11 @@ else if (destination.segmentCount() == 1)
 		}
 
 		// update filters in project descriptions
-		if (source.getProject().exists() && source instanceof Container && ((Container) source).hasFilters()) {
+		// copy filters for any container except project. In case of project copy/move
+		// description file is completely copied/moved. Otherwise it creates the issue
+		// #381
+		if (source.getProject().exists() && source instanceof Container && ((Container) source).hasFilters()
+				&& !movingProject) {
```

Comment:
I think `!movingProject` is not needed within the if block. I will remove and check it.
In case of project move/copy we are copying each item of the project including `description file` later stage. This step copies the filters also(Other content also).
This code is needed for copying filters when we move a folder which has filters on it(not on project node)  from one project to another project. As we don't copy entire description file so we need to copy filters here.

Reply:
**where** is "later stage"?

Reply:
Project filters are copied at,
`Project#internalCopy(IProjectDescription destDesc, int updateFlags, IProgressMonitor monitor)`
`ResourceTree#movedProjectSubtree(IProject project, IProjectDescription destDescription)`